bundles/org.eclipse.equinox.p2.repository/src/org/eclipse/equinox/internal/p2/repository/helpers/AbstractRepositoryManager.java

```diff
@@ -739,12 +739,21 @@ private LocationProperties loadIndexFile(URI location, IProgressMonitor monitor)
 	private LocationProperties handleRemoteIndexFile(URI indexFileURI, IProgressMonitor monitor) {
 		ByteArrayOutputStream index = new ByteArrayOutputStream();
 		IStatus indexFileStatus = null;
-		indexFileStatus = getTransport().download(indexFileURI, index, monitor);
-		while (indexFileStatus.getCode() == IArtifactRepository.CODE_RETRY) {
-			indexFileStatus = getTransport().download(indexFileURI, index, monitor);
+		Transport transport = getTransport();
+		if (transport != null) {
+			indexFileStatus = transport.download(indexFileURI, index, monitor);
+			while (indexFileStatus.getCode() == IArtifactRepository.CODE_RETRY) {
+				indexFileStatus = transport.download(indexFileURI, index, monitor);
+			}
+			if (indexFileStatus != null && indexFileStatus.isOK())
+				return LocationProperties.create(new ByteArrayInputStream(index.toByteArray()));
+		} else {
+			try (InputStream openStream = indexFileURI.toURL().openStream()) {
```

Comment:
My big concern here is that this masks a severe configuration problem. A problem likely to crop up elsewhere and a problem that, if not fixed will, will disable all kinds of functionality such as proxy support.   Surely if one gets the index, something downstream will fail instead.

It would seem better to fail with some kind of message about the transport not being configured available and the like cause(s) for that.